### PR TITLE
Api800/logical interconnect groups

### PR DESCRIPTION
**Pull request by dileepds**
**Description**
API 800 Support for LIG

**Check List**

- [ ] New functionality includes testing.
- [X]  All tests pass for go 1.11 + gofmt checks .
- [ ] New functionality has been documented in the README if applicable.
- [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] New endpoints supported are updated in the endpoints-support.md file.
- [ ]  Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/logical_interconnect_groups.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	fmt.Println(lig_ds)
+
+	fmt.Println("... Updating LogicalInterconnectGroup ...")
+	fmt.Println("")
+	lig_uri.Name = new_lig_name
+	err := ovc.UpdateLogicalInterconnectGroup(lig_uri)
+	if err != nil {
+		panic(err)
+	}
+	if err == nil {
```
Comment (madhav-bharadwaj) [inline examples/logical_interconnect_groups.go]: Same as above. Can an else be used?

**Comment by madhav-bharadwaj** [comment]
Also, i think we should add support for the other query parameters also for the getAll call.

File: examples/logical_interconnect_groups.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		fmt.Println(".....Updated Logical Interconnect Group Successfully....")
+	}
+	fmt.Println("... Deleting LogicalInterconnectGroup ...")
+	del_err := ovc.DeleteLogicalInterconnectGroup(lig_uri.Name)
+	if del_err != nil {
+		panic(del_err)
+	}
+	if del_err == nil {
```
Comment (madhav-bharadwaj) [inline examples/logical_interconnect_groups.go]: else should be used.

File: endpoints-support.md [hunk 3]
```diff
@@ -62,12 +62,13 @@
...
+|<sub>/rest/logical-interconnect-groups</sub>                                             | POST     | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |  :heavy_multiplication_x: | :white_check_mark:   |
+|<sub>/rest/logical-interconnect-groups/{id}</sub>                                        | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |  :heavy_multiplication_x: | :white_check_mark:   |
+|<sub>/rest/logical-interconnect-groups/defaultSettings</sub>                                        | GET      | :heavy_multiplication_x:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   |  :heavy_multiplication_x: | :white_check_mark:   |
+|<sub>/rest/logical-interconnect-groups/{id}</sub>                                        | PUT      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |  :heavy_multiplication_x: |  :white_check_mark:   |
+|<sub>/rest/logical-interconnect-groups/{id}</sub>                                        | PATCH    | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_multiplication_x:   |  :heavy_multiplication_x: |  :heavy_minus_sign:  |
```
Comment (soodpr) [inline endpoints-support.md]: you can add minus sign for API600 as well for PATCH

**Comment by soodpr** [comment]
@dileepds - for GetLogicalInterconnectGroups, we can add start and count query params

**Pull request merged**